data/pigui/modules/new-game-window/class.lua

```diff
@@ -170,6 +170,7 @@ local function startGame(gameParams)
 
 	-- XXX horrible hack here to avoid paying a spawn-in docking fee
 	player:setprop("is_first_spawn", true)
+	FlightLog.SkipFirstDocking()
```

Comment:
I think this problem, as well as the one that is repaired by the previous line in a similar style, should not be repaired in this way. We just need to not generate a docking event when the game starts.

Reply:
If you're able to find a way to cleanly plumb that through the SpaceStation API, I'd love to merge something to that regard. This is definitely a horrible hack, as you can see from the code fragment :smile:

Reply:
Yeah, I totally agree.  I didn't want to get into too many tendrils of changes here but did want to fix the immediate issue with the flightlog.

Perhaps I should create a bug and then can resolve this separately and remove the hacks?

Reply:
I think yes, this problem should be solved separately. I would like to note that in #5619 the "state machine" that controls docking changes quite a lot, so I think it’s worth postponing the solution to this problem until the fate of the specified PR is decided.

Reply:
I will get to reviewing that PR sometime hopefully today.